### PR TITLE
4.x: Update streaming example so it starts and downloads last uploaded file

### PR DESCRIPTION
**Pull request by barchetta**
### Description

This updates the webserver streaming example to:

1. Correct the README which still referred to reactive webserver APIs
2. Fix it so it can start by NOT using a zip file URI to access the download file (see #8477)
3. Have the upload set the file to download as described by the README

### Documentation

README has been updated

### CODE REVIEW

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/Main.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
  */
 public class Main {
 
-    static final String LARGE_FILE_RESOURCE = "/large-file.bin";
+    static final String LARGE_FILE_PATH = "target/classes/large-file.bin";
```
Comment (romain-grecourt) [inline examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/Main.java:28]: This assumes the current working directory is the example's directory.
How about we don't have a default download file, and in order to exercise the example you must first upload a file ?

We can provide `large-file.bin` as a specimen to be used for the steps. We 'd put the file at the top-level next to `README.md`.

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java [hunk 2]
```diff
@@ -38,14 +38,13 @@
...
+    private volatile File file;
 
     StreamingService() {
-        try {
-            filePath = Paths.get(Objects.requireNonNull(getClass().getResource(Main.LARGE_FILE_RESOURCE)).toURI());
-        } catch (URISyntaxException e) {
-            throw new RuntimeException(e);
-        }
+        // Default download file;
+        file = Paths.get(Main.LARGE_FILE_PATH).toFile();
```
Comment (romain-grecourt) [inline examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java:47]: If we implement the suggestion above, we need the file to not exist.
Thus we can't use `File.createTemporaryFile`, instead we can resolve a file path in a temporary directory.

I.e. 

```java
try {
   file = Files.createTempDirectory("helidon-webserver-streaming-example").resolve("download.bin");
} catch (IOException e) {
    throw new UncheckedIOException(e);
}
```

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java [hunk 3]
```diff
@@ -38,14 +38,13 @@
 public class StreamingService implements HttpService {
     private static final Logger LOGGER = Logger.getLogger(StreamingService.class.getName());
 
-    private final Path filePath;
+    // Last file uploaded (or default). Since we don't do any locking
+    // when operating on the file this example is not safe for concurrent requests.
+    private volatile File file;
```
Comment (romain-grecourt) [inline examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java:43]: We should use `java.nio.file.Path`.

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java [hunk 4]
```diff
@@ -59,22 +58,28 @@ private void upload(ServerRequest request, ServerResponse response) {
...
             throw new UncheckedIOException(e);
         }
-        LOGGER.info("Exiting upload ...");
+        LOGGER.info("Exiting upload after uploading " + file.length() + " bytes...");
     }
 
     private void download(ServerRequest request, ServerResponse response) {
         LOGGER.info("Entering download ..." + Thread.currentThread());
-        long length = filePath.toFile().length();
+        if (!file.canRead()) {
```
Comment (romain-grecourt) [inline examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java:71]: `Files.isReadable()`

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/Main.java (hunk 1 above)
Comment (barchetta) [reply examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/Main.java]: Done

Comment (barchetta) [reply examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/Main.java]: Done

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java (hunk 3 above)
Comment (barchetta) [reply examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java]: Done

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java (hunk 4 above)
Comment (barchetta) [reply examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java]: Removed this check

**Review by romain-grecourt — APPROVED** [review]
Nice!

File: examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java (hunk 2 above)
Comment (barchetta) [reply examples/webserver/streaming/src/main/java/io/helidon/examples/webserver/streaming/StreamingService.java]: The code, as written, creates a new temp file every time.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**